drivers/sensor/lis2dw12/lis2dw12.c

```diff
@@ -292,8 +292,9 @@ static int lis2dw12_init(const struct device *dev)
 		return -EIO;
 	}
 
-	/* set default odr and full scale for acc */
-	if (lis2dw12_data_rate_set(lis2dw12->ctx, LIS2DW12_DEFAULT_ODR) < 0) {
+	/* set default odr to 12.5Hz acc */
+	if (lis2dw12_set_odr(dev, 12) < 0) {
+		LOG_ERR("odr init error (12.5 Hz)");
 		return -EIO;
 	}
```

Comment:
better propagate error code here.

Reply:
ok, sure

Reply:
Actually I saw there are other places where it is better to return what the API is returning. So, I'm not changing the commit itselfm, but I'm adding an extra commit on top of everything.

Reply:
thanks!